### ISSUE TITLE
Azure CS:  No space left on device during mounting default token

### ISSUE THREAD

**Issue by olostan** [kubernetes/kubernetes#39508]
**Kubernetes version** (use `kubectl version`):
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.6", GitCommit:"e569a27d02001e343cb68086bc06d47804f62af6", GitTreeState:"clean", BuildDate:"2016-11-12T05:22:15Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.6", GitCommit:"e569a27d02001e343cb68086bc06d47804f62af6", GitTreeState:"clean", BuildDate:"2016-11-12T05:16:27Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: Azure CS
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04 LTS 
- **Kernel** (e.g. `uname -a`): 4.4.0-57-generic 
- **Install tools**: Azure cli (2 agents, standard d2)

**What happened**:
Everything were working, until I decided to update deployment (using `replace`). After that pods stop starting and staid on `ContainerCreating` status.
In 'kubectl describe pod` I got:
```
  28m		40s		22	{kubelet k8s-agent-941b4238-0}			Warning		FailedMount	MountVolume.SetUp failed for volume "kubernetes.io/secret/71af8ad3-d3c6-11e6-9251-000d3a216b93-default-token-i9xv0" (spec.Name: "default-token-i9xv0") pod "71af8ad3-d3c6-11e6-9251-000d3a216b93" (UID: "71af8ad3-d3c6-11e6-9251-000d3a216b93") with: mount failed: exit status 32
Mounting arguments: tmpfs /var/lib/kubelet/pods/71af8ad3-d3c6-11e6-9251-000d3a216b93/volumes/kubernetes.io~secret/default-token-i9xv0 tmpfs []
Output: mount: mount tmpfs on /var/lib/kubelet/pods/71af8ad3-d3c6-11e6-9251-000d3a216b93/volumes/kubernetes.io~secret/default-token-i9xv0 failed: No space left on device


  26m	23s	13	{kubelet k8s-agent-941b4238-0}		Warning	FailedMount	Unable to mount volumes for pod "router-4284723421-h9de2_default(71af8ad3-d3c6-11e6-9251-000d3a216b93)": timeout expired waiting for volumes to attach/mount for pod "router-4284723421-h9de2"/"default". list of unattached/unmounted volumes=[default-token-i9xv0]
  26m	23s	13	{kubelet k8s-agent-941b4238-0}		Warning	FailedSync	Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "router-4284723421-h9de2"/"default". list of unattached/unmounted volumes=[default-token-i9xv0]
```

However, on agents there is definitely enough space:
```
Filesystem      Size  Used Avail Use% Mounted on
udev            3.4G     0  3.4G   0% /dev
tmpfs           697M   17M  680M   3% /run
/dev/sda1        29G  7.6G   21G  27% /
tmpfs           3.5G     0  3.5G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.5G     0  3.5G   0% /sys/fs/cgroup
none             64K     0   64K   0% /etc/network/interfaces.dynamic.d
/dev/sdb1        14G   40M   14G   1% /mnt
tmpfs           100K     0  100K   0% /run/lxcfs/controllers
tmpfs           3.5G   12K  3.5G   1% /var/lib/kubelet/pods/b7abae8f-cefc-11e6-9251-000d3a216b93/volumes/kubernetes.io~secret/default-token-zk98l
shm              64M     0   64M   0% /var/lib/docker/containers/70f1efbd4a5564208ca79a329dfc3828f0315aba5d7b5034b6784e9341e96a10/shm
tmpfs           3.5G   12K  3.5G   1% /var/lib/kubelet/pods/9c1dfead-d393-11e6-9251-000d3a216b93/volumes/kubernetes.io~secret/default-token-i9xv0
shm              64M     0   64M   0% /var/lib/docker/containers/b050aafce252bb14f65db18315ff2957ae3c08854806a65693d96534749f9d73/shm
shm              64M     0   64M   0% /var/lib/docker/containers/7643840c8ea7465697ccbc3966a6ce7d5a6b852ae924787b1892945458e4695e/shm
tmpfs           697M     0  697M   0% /run/user/1000
```
(I even see that token was mounted)

It looks very close to https://github.com/kubernetes/kubernetes/issues/13174 but seems it is somehow Azure related...

Also on same agent I have another pod running with a volume mounted:
```
Volumes:
  default-token-i9xv0:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-i9xv0
```

**Comment by olostan** [comment]
Small note: assigning pod to another node solved the problem. However i can't identify any difference between nodes.

**Comment by brendandburns** [comment]
There were lots of fixes to disk mount/unmount in 1.5.x and 1.6.x, I'm going to close this, if it re-occurs, please re-open.
